client/src/schema/query.ts

```diff
@@ -201,24 +199,37 @@ export class Query<T extends XataRecord, R extends XataRecord = T> implements Pa
    */
   async getMany<Options extends QueryOptions<T>>(
     options: Options = {} as Options
-  ): Promise<
-    (typeof options extends { columns: SelectableColumn<T>[] } ? Select<T, typeof options['columns'][number]> : R)[]
-  > {
+  ): Promise<GetWithColumnOptions<T, R, typeof options>[]> {
     const { records } = await this.getPaginated(options);
     return records;
   }
 
+  /**
+   * Performs the query in the database and returns all the results.
+   * Warning: If there are a large number of results, this method can have performance implications.
+   * @param options Additional options to be used when performing the query.
+   * @returns An array of records from the database.
+   */
+  async getAll<Options extends QueryOptions<T>>(
+    options: Omit<Options, 'page'> = {} as Options
+  ): Promise<GetWithColumnOptions<T, R, typeof options>[]> {
+    const results = [];
+
+    for await (const page of this.getIterator(100, options)) {
```

Comment:
Could we extract this number to a constant?
Question: is 100 the limit of the API?

Reply:
Actually no, moving discussion to #108